manager/dispatcher/dispatcher.go

```diff
@@ -116,6 +116,17 @@ func (b weightedPeerByNodeID) Len() int { return len(b) }
 
 func (b weightedPeerByNodeID) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
 
+// tasksEqual returns true if the tasks are functionaly equal, ignoring status,
+// version and other superfluous fields.
+func tasksEqual(a, b *api.Task) bool {
+	a, b = a.Copy(), b.Copy()
```

Comment:
Because soon, there's going to be lots of intertwined garbage into that package. Just a matter of time.